[21970] Add definingProject as an optgroup to version dropdown

oliverguenther opened the pull request:
Versions should be grouped by project in the inplace edit of versions.
This provides this functionality as an optgroup through the `definingProject` embedded property. 

Adding the same functionality for work package filters is out of scope for this PR.

Relevant work package: https://community.openproject.org/work_packages/21970/activity

oliverguenther committed:
Add definingProject as group to version

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-21970-version-hierarchy.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-21970-version-hierarchy.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Extract groups from options when they are embedded

We return the actual embedded resource as allowedValues when they are available.
As there are drop-down lists where no embedded information is given
(e.g., assignee, responsible), we conditionally extract group
information from embedded resources.

oliverguenther committed:
Add version spec
